### DISCUSSION TITLE
Allgemeine TileCaches - Dateisystem-Pfad für Tiles

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#48] [General]
Hallo Jürgen!

Tritt bei mir mit Version 5.23.901 auf.
Wir haben einen ausschließlich intern verfügbaren TileCache im CMS konfiguriert. Einerseits die URL für die Tiles, andererseits den optionalen Parameter "Dateisystem-Pfad für Tiles" (siehe Screenshot).

![tc_geologie_cms](https://user-images.githubusercontent.com/122986367/225330959-5a3a2009-5ba0-4d75-921d-c4620adc6d9d.png)

Der TileCache wird in der Karte problemlos dargestellt, beim Drucken gibt es allerdings einen Fehler. Der Fehler tritt ausschließlich auf, wenn im Parameter "Dateisystem-Pfad für Tiles" ein Wert eingetragen ist. 

![tc_geologie_print](https://user-images.githubusercontent.com/122986367/225331604-a8b2786e-259f-467c-816e-22cce938fab0.png)

Die Konfiguration im webigs5 ist 1:1 zur Konfiguration im webgis4. Beide Instanzen laufen am gleichen Server. Im webgis4 funktioniert diese Konfig problemlos. 
Ist im webgis5 bei der Konfiguration irgendwas noch zu beachten/zu ändern? Warum wird dieser Fehler beim Drucken ausgegeben?

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Hallo Andi,

Ein paar Fragen hätte ich zu diesem Thema:
1. Ist es für diesen Dienst notwendig, dass man einen Datasystem-Pfad angibt? Diese Option kommt eigentlich noch aus der WebGIS-Steinzeit. Eigentlich möchte ich das los haben.
2. Hat WebGIS überhaupt Zugriff auf das Dateisystem Verzeichnis?
3. Muss man die *Url für Tiles* unbedingt relativ angeben?
4. Ist das etwas, das ich mir einmal "live" anschauen kann?

LG Jürgen

**Comment by gis-eni** [comment]
Ok, nach drüber schauen, bin ich noch zu folgender Erkenntnis gekommen:

Für (webGIS 4) konnte in der Config ein ImpersonateUser definiert werden. Das war ein User, der das Rechte hatte auf gewisse File System Pfade zuzugreifen. Der WebGIS Prozess hat sich dann bei Bedarf als dieser User verkleidet und die Daten aus dem Filesystem angeholt.

Diese Technologie ist mittlerweile Geschichte und die Vorgehensweise ist die, das der ApplikationPool User die Rechte hat, die er braucht.
Die Allgemeinen TileCaches mit Pfad sind die letzte Teil, wo beim Drucken noch auf den ImpersonationContext zuzugreifen wird und das verursacht diese NullReferenceException. Das ist leider unabhängig davon, ob du einen ImpersonateUser in der config eingetragen hast oder nicht :(

Dieser Bug ist (erst) mit dem nächsten Release (>=5.23.1101) behoben.

Man sollte sich trotzdem die Frage stellen, warum man nicht gleich über eine absolute Url auf die Tiles zugreift. Damit erspart man sich, dass der AppPool User lesereichte auf ein Fileshare benötigt.

LG Jürgen
